jdk_distribution_parser.py

```diff
@@ -0,0 +1,171 @@
+#
+# ----------------------------------------------------------------------------------------------------
+#
+# Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+#
+import re
+import mx
+from os.path import join
+
+GITHUB_URL = "https://github.com"
+GITHUB_RELEASES = "releases/download"
+
+class JdkDistribution(object):
+    _jdk_distributions = []
+    _DEFAULT_JDK = "labsjdk-ce-11"
+
+    @staticmethod
+    def parse(name, version):
+        for dist in JdkDistribution.__subclasses__():
+            if dist._name == name:
+                dist(version)
+
+    @staticmethod
+    def choose_dist(quiet=False):
+        if quiet:
+            return JdkDistribution.by_name(JdkDistribution._DEFAULT_JDK)
+
+        index = 1
+        default_choice = 1
+        for dist in JdkDistribution._jdk_distributions:
+            if dist.get_name() == JdkDistribution._DEFAULT_JDK:
+                default_choice = index
+                default = "*"
+            else:
+                default = " "
+            print("[{index}]{default} {name} | {version}".format(index=index,
+            name=dist.get_name().ljust(25), version=dist.get_version(), default=default))
+            index += 1
+        while True:
+            print("Select JDK>"), # pylint: disable=expression-not-assigned
```

Comment:
`input` should be used here instead of `print` so that `Select JDK>` is a prompt on the same line that the user provides their input.

Reply:
Fixed